drivers/rtc/max31343/max31343.h

```diff
@@ -0,0 +1,194 @@
+#define MAX31343_R_CFG1				0x03
+#define MAX31343_R_CFG2				0x04
+#define MAX31343_R_TIMER_CONFIG			0x05
+#define MAX31343_R_SECONDS			0x06
+#define MAX31343_R_MINUTES			0x07
+#define MAX31343_R_HOURS			0x08
+#define MAX31343_R_DAY				0x09
+#define MAX31343_R_DATE				0x0A
+#define MAX31343_R_MONTH			0x0B
+#define MAX31343_R_YEAR				0x0C
+#define MAX31343_R_ALM1_SEC			0x0D
+#define MAX31343_R_ALM1_MIN			0x0E
+#define MAX31343_R_ALM1_HRS			0x0F
+#define MAX31343_R_ALM1DAY_DATE			0x10
+#define MAX31343_R_ALM1_MON			0x11
+#define MAX31343_R_ALM1_YEAR			0x12
+#define MAX31343_R_ALM2_MIN			0x13
+#define MAX31343_R_ALM2_HRS			0x14
+#define MAX31343_R_ALM2DAY_DATE			0x15
+#define MAX31343_R_TIMER_COUNT			0x16
+#define MAX31343_R_TIMER_INIT			0x17
+#define MAX31343_R_PWR_MGMT			0x18
+#define MAX31343_R_TRICKLE			0x19
+#define MAX31343_R_TEMP_MSB			0x1A
+#define MAX31343_R_TEMP_LSB			0x1B
+#define MAX31343_R_TS_CONFIG			0x1C
+#define MAX31343_R_RAM_REG_START		0x22
+#define MAX31343_R_RAM_REG_END			0x61
+
+/* Status register bits */
+#define MAX31343_F_STATUS_A1F			NO_OS_BIT(0)
+#define MAX31343_F_STATUS_A2F			NO_OS_BIT(1)
+#define MAX31343_F_STATUS_TIF			NO_OS_BIT(2)
+#define MAX31343_F_STATUS_TSF			NO_OS_BIT(3)
+#define MAX31343_F_STATUS_PFAIL			NO_OS_BIT(5)
+#define MAX31343_F_STATUS_OSF			NO_OS_BIT(6)
+#define MAX31343_F_STATUS_PSDECT		NO_OS_BIT(7)
+
+/* Interrupt enable register bits */
+#define MAX31343_F_INT_EN_A1IE			NO_OS_BIT(0)
+#define MAX31343_F_INT_EN_A2IE			NO_OS_BIT(1)
+#define MAX31343_F_INT_EN_TIE			NO_OS_BIT(2)
+#define MAX31343_F_INT_EN_TSIE			NO_OS_BIT(3)
+#define MAX31343_F_INT_EN_PFAILE		NO_OS_BIT(5)
+#define MAX31343_F_INT_EN_DOSF			NO_OS_BIT(6)
+
+/* RTC reset register bits */
+#define MAX31343_F_RTC_RESET_SWRST		NO_OS_BIT(0)
+
+/* Config 1 register bits */
+#define MAX31343_F_CFG1_ENOSC			NO_OS_BIT(1)
+#define MAX31343_F_CFG1_I2C_TIMEOUT		NO_OS_BIT(3)
+#define MAX31343_F_CFG1_DATA_RET		NO_OS_BIT(4)
+
+/* Config 2 register bits */
+#define MAX31343_F_CFG2_SQW_HZ			NO_OS_BIT(0)
+#define MAX31343_F_CFG2_CLKO_HZ			NO_OS_BIT(3)
+#define MAX31343_F_CFG2_ENCLKO			NO_OS_BIT(7)
+
+/* Timer config register bits */
+#define MAX31343_F_TIMER_CONFIG_TFS		NO_OS_BIT(0)
+#define MAX31343_F_TIMER_CONFIG_TRPT		NO_OS_BIT(2)
+#define MAX31343_F_TIMER_CONFIG_TPAUSE		NO_OS_BIT(3)
+#define MAX31343_F_TIMER_CONFIG_TE		NO_OS_BIT(4)
+
+/* Power management register bits */
+#define MAX31343_F_PWR_MGMT_DMAN_SEL		NO_OS_BIT(2)
+#define MAX31343_F_PWR_MGMT_D_VBACK_SEL		NO_OS_BIT(3)
+#define MAX31343_F_PWR_MGMT_PFVT		NO_OS_BIT(4)
+
+/* Trickle register bits */
+#define MAX31343_F_TRICKLE_D_TRICKLE		NO_OS_BIT(0)
+#define MAX31343_F_TRICKLE_TCHE			NO_OS_BIT(4)
+
+/* Temperature sensor config register bits */
+#define MAX31343_F_TS_CONFIG_TTSINT		NO_OS_BIT(3)
+#define MAX31343_F_TS_CONFIG_ONESHOT_MODE	NO_OS_BIT(6)
+#define MAX31343_F_TS_CONFIG_AUTO_MODE		NO_OS_BIT(7)
+
+#define MAX_31343_BCD2BIN(val)			(((val) & 15) + ((val) >> 4) * 10)
```

Comment:
these macros should be generic. we don't want to have them duplicated in each rtc driver.
See:
https://github.com/torvalds/linux/blob/master/include/linux/bcd.h

Reply:
Is it ok if I move these 2 defines into: no_os_util.h